### ISSUE TITLE
Pasted non-block level text before a list/heading becomes a list item

### ISSUE THREAD

**Issue by borodean** [quilljs/quill#1338]
Pasted non-block level text before the list becomes a list item.

**Steps for Reproduction:**
1. Visit [the playgorund](http://quilljs.com/playground/).
2. Go to the HTML tab.
3. Paste this inside the `editor-container`:

```html
List:
<ul>
  <li>1</li>
  <li>2</li>
  <li>3</li>
</ul>
```

**Expected behavior**:
First line becomes a paragraph.

**Actual behavior**:
First line becomes an item of the following list.

**Platform**:
Chrome 56, Windows 10

**Version**:
1.2.2

**Comment by borodean** [comment]
Same happend with headings. Paste this:

```html
Before the list
<h2>Heading 2</h2>
```

And you'll get the first line formatted as an another `h2`.

**Comment by jhchen** [comment]
This has been fixed in 1.2.4.
